doc/stdenv/cross-compilation.chapter.md

```diff
@@ -78,21 +78,46 @@ If both the dependency and depending packages aren't compilers or other machine-
 
 Finally, if the depending package is a compiler or other machine-code-producing tool, it might need dependencies that run at "emit time". This is for compilers that (regrettably) insist on being built together with their source languages' standard libraries. Assuming build != host != target, a run-time dependency of the standard library cannot be run at the compiler's build time or run time, but only at the run time of code emitted by the compiler.
 
-Putting this all together, that means we have dependencies in the form "host → target", in at most the following six combinations:
+Putting this all together, that means that we have dependency types of the form "X→ E", which means that the dependency executes on X and emits code for E; each of X and E can be `build`, `host`, or `target`, and E can be `*` to indicate that the dependency is not a compiler-like package.
+
+Dependency types describe the relationships that a package has with each of its transitive dependencies.  You could think of attaching one or more dependency types to each of the formal parameters at the top of a package's `.nix` file, as well as to all of *their* formal parameters, and so on.   Triples like `(foo, bar, baz)`, on the other hand, are a property of an instantiated derivation -- you could would attach a triple `(mips-linux, mips-linux, sparc-solaris)` to a `.drv` file in `/nix/store`.
+
+Only nine dependency types matter in practice:
 
 #### Possible dependency types {#possible-dependency-types}
 
-| Dependency’s host platform | Dependency’s target platform |
-|----------------------------|------------------------------|
-| build                      | build                        |
-| build                      | host                         |
-| build                      | target                       |
-| host                       | host                         |
-| host                       | target                       |
-| target                     | target                       |
+| Dependency type     | Dependency’s host platform | Dependency’s target platform |
+|---------------------|----------------------------|------------------------------|
+| build→ *       | build                      | (none)                       |
+| build→ build   | build                      | build                        |
+| build→ host    | build                      | host                         |
+| build→ target  | build                      | target                       |
+| host→ *        | host                       | (none)                       |
+| host→ host     | host                       | host                         |
+| host→ target   | host                       | target                       |
+| target→ *      | target                     | (none)                       |
+| target→ target | target                     | target                       |
```

Comment:
```suggestion
| Dependency type | Dependency’s host platform | Dependency’s target platform |
|-----------------|----------------------------|------------------------------|
| build→ *        | build                      | (none)                       |
| build→ build    | build                      | build                        |
| build→ host     | build                      | host                         |
| build→ target   | build                      | target                       |
| host→ *         | host                       | (none)                       |
| host→ host      | host                       | host                         |
| host→ target    | host                       | target                       |
| target→ *       | target                     | (none)                       |
| target→ target  | target                     | target                       |
```

Reply:
Thanks, my terminal (alacritty) still doesn't seem to handle unicode properly 100% of the time for some reason.